compiler/natives/src/net/http/fetch.go

```diff
@@ -116,7 +117,7 @@ func (t *fetchTransport) RoundTrip(req *Request) (*Response, error) {
 		},
 		func(reason *js.Object) {
 			select {
-			case errCh <- errors.New("net/http: fetch() failed"):
+			case errCh <- fmt.Errorf("net/http: fetch() failed: %s", reason.String()):
```

Comment:
I guess it should never be `nil` for the same reason `result` is never nil in the success case.

I looked at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then, and don't see an explicit confirmation that reason won't be nil...

But maybe it would be? Is there a way we can be sure?